bin/jsondiff.py

```diff
@@ -20,54 +31,48 @@ def add_arguments():
                          help='Specify the maximum relative error (default 1e-8)',
                          type=float,
                          nargs='?', 
-                         const=1e-8)
+                         const=1e-8,
+                         default=None)
     options.add_argument('--abs_tol',
                          metavar='<absolute_error>',
                          dest='absolute_error',
                          help='Specify the maximum absolute error (default 1e-10)',
                          type=float,
                          nargs='?', 
-                         const=1e-10)
+                         const=1e-10,
+                         default=None)
     options.add_argument('-h', '--help', action='help', help='Displays CLI usage statement')
     args, unknown = parser.parse_known_args()
-    
-    # Set arg parameter to None if no flag was supplied
-    if '--rel_tol' not in sys.argv[1:]:
-        args.relative_error = None
-    if '--abs_tol' not in sys.argv[1:]:
-        args.absolute_error = None
     return args
 
-def validate_json_extension(json_file):
-    """Validates that the json file has an .json extension and path exists
-        json_file: the file to validate"""
-    return_code = 0
-    base, extension = os.path.splitext(json_file)
-    if extension.lower() not in ('.json'):
-        raise argparse.ArgumentTypeError('Invalid extension with "' + str(json_file) + '": Provide a file with a .json extension')
-    return json_file
-
-def validate_json_files_exist(args):
-    """Check if json files exist in the system path
-        args: arguments provided by the user"""
-    return_code = 0
-    are_files = list()
-    for json_file in args.json_files:
-        if not os.path.exists(json_file):
-            print('Path does not exist: ' + str(json_file))
-            return_code += 1
-    return return_code
-
-def validate_flags(args):
+def validate_flags(args, raise_on_error=False, log_on_error=True):
     """Validate that only one flag was passed via the command-line interface
-        args: arguments provided by the user"""
+        args: arguments provided by the user
+        raise_on_error: raises a ValueError if invalid quantity of tolerance flags
+        log_on_error: reports the error in a log file"""
     return_code = 0
+    log = logging.getLogger(__name__)
+
+    # When no flags are provided
     if args.relative_error is None and args.absolute_error is None:
         return_code += 1
-        print('Error: No flag provided. Use "-h" or "--help" for more information.')
+        help = 'Use "-h" or "--help" for more information.'
+        error = 'No flag provided.'
+        message = '{0}: {1}'.format('ValueError', error)
+        if log_on_error:
+            log.error(message)
+        if raise_on_error:
+            raise ValueError(error + ' ' + help)
```

Comment:
I would prefer you use `format` everywhere, I am trying to get more consistent, even if it is a bit longer.

Reply:
What do you mean by `format`? Do you mean when I create string do `.format()`?